dangerzone/gui/main_window.py

```diff
@@ -585,6 +631,101 @@ def dangerous_doc_button_clicked(self) -> None:
             pass
 
 
+class DocSelectionDropFrame(QtWidgets.QFrame):
+    """
+    HACK Docs selecting widget "drag-n-drop" border widget
+    The border frame doesn't show around the whole widget
+    unless there is another widget wrapping it
+    """
+
+    documents_selected = QtCore.Signal(list)
+
+    def __init__(
+        self, dangerzone: DangerzoneGui, docs_selection_widget: DocSelectionWidget
+    ) -> None:
+        super().__init__()
+
+        self.dangerzone = dangerzone
+        self.docs_selection_widget = docs_selection_widget
+
+        # Drag and drop functionality
+        self.setAcceptDrops(True)
+
+        self.document_image_text = QtWidgets.QLabel(
+            "Drag and drop\ndocuments here\n\nor"
+        )
+        self.document_image_text.setAlignment(QtCore.Qt.AlignCenter)
+        self.document_image = QtWidgets.QLabel()
+        self.document_image.setAlignment(QtCore.Qt.AlignCenter)
+        self.document_image.setPixmap(
+            load_svg_image("document.svg", width=20, height=24)
+        )
+
+        self.center_layout = QtWidgets.QVBoxLayout()
+        self.center_layout.addWidget(self.document_image)
+        self.center_layout.addWidget(self.document_image_text)
+        self.center_layout.addWidget(self.docs_selection_widget)
+
+        self.drop_layout = QtWidgets.QVBoxLayout()
+        self.drop_layout.addStretch()
+        self.drop_layout.addLayout(self.center_layout)
+        self.drop_layout.addStretch()
+
+        self.setLayout(self.drop_layout)
+
+    def dragEnterEvent(self, ev: QtGui.QDragEnterEvent) -> None:
+        ev.accept()
+
+    def dragLeaveEvent(self, ev: QtGui.QDragLeaveEvent) -> None:
+        ev.accept()
+
+    def dropEvent(self, ev: QtGui.QDropEvent) -> None:
+        ev.setDropAction(QtCore.Qt.CopyAction)
+        documents = []
+        supported_exts = get_supported_extensions()
+        for url_path in ev.mimeData().urls():
+            doc_path = url_path.toLocalFile()
+            doc_ext = os.path.splitext(doc_path)[1]
+            if doc_ext in supported_exts:
+                documents += [Document(doc_path)]
```

Comment:
I have tested drag&drop with a file that contains an invalid UTF-8 sequence in its filename (see also #768) and I get the following traceback in the terminal (the extra log line with `url_path` is added by me):
```
[INFO] url_path: PySide6.QtCore.QUrl('file:///tmp/sample%F0.pdf')
Traceback (most recent call last):
  File "/home/naglis/code/dangerzone/dangerzone/document.py", line 66, in validate_input_filename
    open(filename, "rb")
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/sample�.pdf'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/naglis/code/dangerzone/dangerzone/gui/main_window.py", line 691, in dropEvent
    documents += [Document(doc_path)]
                  ^^^^^^^^^^^^^^^^^^
  File "/home/naglis/code/dangerzone/dangerzone/document.py", line 50, in __init__
    self.input_filename = input_filename
    ^^^^^^^^^^^^^^^^^^^
  File "/home/naglis/code/dangerzone/dangerzone/document.py", line 95, in input_filename
    self.validate_input_filename(filename)
  File "/home/naglis/code/dangerzone/dangerzone/document.py", line 68, in validate_input_filename
    raise errors.InputFileNotFoundException() from e
dangerzone.errors.InputFileNotFoundException: Input file not found: make sure you typed it correctly.
```

I do not have a graphical file manager and I used [dragon](https://github.com/mwh/dragon) to have somewhere to drag from, so maybe this is a problem with the source (dragon), but the [docs of `toLocalFile()`](https://doc.qt.io/qt-6/qurl.html#toLocalFile) also mention:

> Note: if the path component of this URL contains a non-UTF-8 binary sequence (such as %80), the behaviour of this function is undefined.

hence it could be that such URLs are simply not supported by Qt :shrug: If you have a GUI file manager at your disposal then perhaps you could test it as well? To create such a file in bash I've used `touch $'sample\xf0.pdf'`

Reply:
Thank you for testing! 

> Was it '/tmp/sample%F0.pdf'?

That was just an example, IIUC it could be any path that contains byte sequence(s) that would be invalid when converting the path to Unicode in Python using the file system encoding (I am assuming UTF-8).

I guess this is something caused by my configuration or perhaps the fact that I've used _dragon_ for the drag&drop. Sorry for wasting your time :flushed: